public/app/plugins/datasource/prometheus/components/PromExploreQueryEditor.tsx

```diff
@@ -1,85 +1,63 @@
-export type Props = ExploreQueryFieldProps<PrometheusDatasource, PromQuery, PromOptions>;
-
-interface State {
-  interval: string;
-}
+import { PromExploreExtraField } from './PromExploreExtraField';
 
-export class PromExploreQueryEditor extends PureComponent<Props, State> {
-  // Query target to be modified and used for queries
-  query: PromQuery;
+export type Props = ExploreQueryFieldProps<PrometheusDatasource, PromQuery, PromOptions>;
 
-  constructor(props: Props) {
-    super(props);
-    const { query } = props;
-    this.query = query;
-    // Query target properties that are fully controlled inputs
-    this.state = {
-      // Fully controlled text inputs
-      interval: query.interval,
-    };
+export const PromExploreQueryEditor = memo(function PromExploreQueryEditor(props: Props) {
+  const { query, data, datasource, history, onChange, onRunQuery } = props;
+  const [step, setStep] = useState(query.interval);
+
+  function onChangeQueryStep(value: string, override?: boolean) {
+    const { query, onChange, onRunQuery } = props;
+    if (onChange) {
+      const nextQuery = { ...query, interval: value };
+      onChange(nextQuery);
+      if (override && onRunQuery) {
+        onRunQuery();
+      }
+    }
   }
 
-  onFieldChange = (query: PromQuery, override?: any) => {
-    this.query.expr = query.expr;
-  };
-
-  onIntervalChange = (e: React.SyntheticEvent<HTMLInputElement>) => {
-    const interval = e.currentTarget.value;
-    this.query.interval = interval;
-    this.setState({ interval });
-  };
-
-  onRunQuery = () => {
-    const { query } = this;
-    this.props.onChange(query);
-    this.props.onRunQuery();
-  };
+  function onStepChange(e: React.SyntheticEvent<HTMLInputElement>) {
+    setStep(e.currentTarget.value);
+  }
 
-  onReturnKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+  function onReturnKeyDown(e: React.KeyboardEvent<HTMLInputElement>) {
     if (e.key === 'Enter') {
-      this.onRunQuery();
+      onRunQuery();
     }
-  };
-
-  render() {
-    const { datasource, query, data, history } = this.props;
-    const { interval } = this.state;
-
-    return (
-      <div className="gf-form-inline">
-        <PromQueryField
-          datasource={datasource}
-          query={query}
-          onRunQuery={this.onRunQuery}
-          onChange={this.onFieldChange}
-          history={history}
-          data={data}
-        >
-          <div className="gf-form-inline explore-input--ml">
-            <div className="gf-form">
-              <FormLabel width={4}>Step</FormLabel>
-              <input
-                type="text"
-                className="gf-form-input width-6"
-                placeholder={'auto'}
-                onChange={this.onIntervalChange}
-                onKeyDown={this.onReturnKeyDown}
-                value={interval}
-              />
-            </div>
-          </div>
-        </PromQueryField>
-      </div>
-    );
   }
-}
+
+  useEffect(() => {
```

Comment:
Nitpick: Seems like we could do this inside the `onChangeFunc` handler. So that we call this directly when the value changes, without having to wait for update.